packages/manager/modules/pci/src/projects/project/public-gateways/edit/edit.routing.js

```diff
@@ -1,10 +1,21 @@
 export default /* @ngInject */ ($stateProvider) => {
   $stateProvider.state('pci.projects.project.public-gateways.edit', {
-    url: '/edit',
+    url: '/edit?gatewayId&region',
     component: 'pciProjectPublicGatewaysEdit',
-    params: { gateway: null },
+    params: {
+      gatewayId: {
+        dynamic: true,
+        type: 'string',
+      },
+      region: {
+        dynamic: true,
+        type: 'string',
+      },
+    },
     resolve: {
-      gateway: /* @ngInject */ ($transition$) => $transition$.params().gateway,
+      gatewayId: /* @ngInject */ ($transition$) =>
+        $transition$.params().gatewayId,
+      region: /* @ngInject */ ($transition$) => $transition$.params().region,
```

Comment:
**question**: Why region is added in the URL? I dont see the need for it.

Reply:
@anooparveti  here I am fetching gateway details by id which needs gatewayId and region to make the api call.